06.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  def initialize(all_roll_result)
+    @frames = build_frames(all_roll_result)
+  end
+
+  def score
+    @frames.sum { |frame| frame.score(@frames) }
+  end
+
+  private
+
+  def build_frames(all_roll_result)
+    frames = []
+    current_frame = nil
+    all_roll_result.split(',').each do |pin|
+      if current_frame.nil? || !current_frame.addable?
+        current_frame = Frame.new(frames.size)
+        frames << current_frame
+      end
+      current_frame.add(pin)
+    end
+    frames
+  end
```

Comment:
```suggestion
  def build_frames(all_roll_result)
    all_roll_result.split(',').each_with_object([]) do |pin, frames|
      frames << Frame.new(frames.size) if !frames.last&.addable?
      frames.last.add(pin)
    end
  end
```

参考）こんな中二病っぽい書き方もできます（真似しなくてもいいですw）

Reply:
サンプルコードありがとうございます。current_frameなしで書くこともできるのですね🤔
each_with_objectは使いどころが多くなさそうですし、少し読みにくく感じるので、今回は修正なしとさせてください。